sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetsOnlyScanConfigBuilder.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.streaming
+
+import org.apache.spark.sql.sources.v2.reader.{ScanConfig, ScanConfigBuilder}
+
+/**
+ * A very simple [[ScanConfigBuilder]] and [[ScanConfig]] implementation that carries offsets for
+ * streaming data sources.
+ */
+case class OffsetsOnlyScanConfigBuilder(start: Offset, end: Option[Offset] = None)
+  extends ScanConfigBuilder with ScanConfig {
```

Comment:
It looks hacky to do so: `extends ScanConfigBuilder with ScanConfig` and return the class itself for `build()`

Reply:
otherwise we need to create 2 very similar classes. I'm fine with both.